homeassistant/components/bosch_alarm/sensor.py

```diff
@@ -0,0 +1,199 @@
+        faults = self.panel.panel_faults
+        return "\n".join(faults) if faults else "No faults"
+
+    async def async_added_to_hass(self) -> None:
+        """Observe state changes."""
+        self.panel.faults_observer.attach(self.schedule_update_ha_state)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Stop observing state changes."""
+        self.panel.faults_observer.detach(self.schedule_update_ha_state)
+
+
+class AreaSensor(SensorEntity):
+    """A faults sensor entity for a bosch alarm panel."""
+
+    _attr_has_entity_name = True
+    _attr_entity_category = EntityCategory.DIAGNOSTIC
+
+    def __init__(self, panel: Panel, area_id: int, unique_id: str, type: str) -> None:
+        """Set up a faults sensor entity for a bosch alarm panel."""
+        self.panel = panel
+        area_unique_id = f"{unique_id}_area_{area_id}"
+        self._area = panel.areas[area_id]
+        self._attr_unique_id = f"{area_unique_id}_{type}"
+        self._attr_translation_placeholders = {"area": self._area.name}
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, area_unique_id)},
+            name=self._area.name,
+            manufacturer="Bosch Security Systems",
+            model=panel.model,
+            sw_version=panel.firmware_version,
+            via_device=(
+                DOMAIN,
+                unique_id,
+            ),
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Observe state changes."""
+        await super().async_added_to_hass()
+        self.panel.faults_observer.attach(self.schedule_update_ha_state)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Stop observing state changes."""
+        self.panel.faults_observer.detach(self.schedule_update_ha_state)
+
+
+class FaultingPointsSensor(AreaSensor):
+    """A faults sensor entity for a bosch alarm panel."""
+
+    _attr_translation_key = "faulting_points"
+
+    def __init__(self, panel: Panel, area_id: int, unique_id: str) -> None:
+        """Set up a faults sensor entity for a bosch alarm panel."""
+        super().__init__(panel, area_id, unique_id, "faults")
+
+    @property
+    def native_value(self) -> str:
+        """The state of this faults entity."""
+        return f"{self._area.faults}"
+
+
+class AreaReadyToArmSensor(AreaSensor):
+    """A sensor entity showing the ready state for an area for a bosch alarm panel."""
+
+    _attr_translation_key = "ready_to_arm"
+
+    def __init__(self, panel: Panel, area_id: int, unique_id: str) -> None:
+        """Set up a faults sensor entity for a bosch alarm panel."""
+        super().__init__(panel, area_id, unique_id, "ready_to_arm")
+
+    @property
+    def native_value(self) -> str:
+        """The state of this entity."""
+        if self._area.all_ready:
+            return "home_and_away_ready"
+        if self._area.part_ready:
+            return "home_ready"
+        return "not_ready"
```

Comment:
I think we agreed on making this a binary sensor right?

Reply:
Ah yeah I was only talking about alarms and panel faults at the time, but I can do this one too